internal/provider/resource_password.go

```diff
@@ -573,154 +578,158 @@ func passwordSchemaV3() tfsdk.Schema {
+					int64planmodifiers.DefaultValue(
+						types.Int64Value(0),
+					),
+					int64planmodifiers.RequiresReplace(),
 				},
 			},
 
-			"min_special": {
+			"min_special": schema.Int64Attribute{
 				Description: "Minimum number of special characters in the result. Default value is `0`.",
-				Type:        types.Int64Type,
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					planmodifiers.DefaultValue(types.Int64Value(0)),
-					planmodifiers.RequiresReplace(),
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifiers.DefaultValue(
+						types.Int64Value(0),
+					),
+					int64planmodifiers.RequiresReplace(),
 				},
 			},
 
-			"override_special": {
+			"override_special": schema.StringAttribute{
 				Description: "Supply your own list of special characters to use for string generation.  This " +
 					"overrides the default character list in the special argument.  The `special` argument must " +
 					"still be set to true for any overwritten characters to be used in generation.",
-				Type:     types.StringType,
 				Optional: true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					resource.RequiresReplaceIf(
-						planmodifiers.RequiresReplaceUnlessEmptyStringToNull(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIf(
+						stringplanmodifiers.RequiresReplaceUnlessEmptyStringToNull(),
 						"Replace on modification unless updating from empty string (\"\") to null.",
 						"Replace on modification unless updating from empty string (`\"\"`) to `null`.",
 					),
 				},
 			},
 
-			"result": {
+			"result": schema.StringAttribute{
 				Description: "The generated random string.",
-				Type:        types.StringType,
 				Computed:    true,
 				Sensitive:   true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					resource.UseStateForUnknown(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
 
-			"bcrypt_hash": {
+			"bcrypt_hash": schema.StringAttribute{
 				Description: "A bcrypt hash of the generated random string.",
-				Type:        types.StringType,
 				Computed:    true,
 				Sensitive:   true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					resource.UseStateForUnknown(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
 
-			"id": {
+			"id": schema.StringAttribute{
 				Description: "A static value used internally by Terraform, this should not be referenced in configurations.",
 				Computed:    true,
-				Type:        types.StringType,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					resource.UseStateForUnknown(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
 		},
 	}
 }
 
+// TODO: passwordSchemaV2 needs to be updated to use schema.Schema once resource.StateUpgrader has been
+// updated to use schema.Schema for PriorSchema.
```

Comment:
Aye, my bad for missing this -- cross reference: https://github.com/hashicorp/terraform-plugin-framework/issues/572

We can certainly handle this TODO after framework v1.0.0 is out.

Reply:
Sounds good to me 👍